protobuf_serialization/pb_option.nim

```diff
@@ -0,0 +1,28 @@
+type PBOption*[T] = object
```

Comment:
How about the following definition for `PBOption`:

```nim

type PBOption*[defaultValue: static[auto]] = object
  some: bool
  value: typeof(defaultValue)

type
  SomeRecord = object
    x: int

type
  T1 = PBOption[10]
  T2 = PBOption["Test"]
  T3 = PBOption[SomeRecord(x: 10)]

proc get*(opt: PBOption): auto =
  if opt.some:
    result = opt.value
  else:
    result = opt.defaultValue

var x1: T1
var x2 = T2(some: true, value: "test")
var x3: T3

echo x1.get
echo x2.get
echo x3.get
```

Reply:
Looks good to me. I can get this edit done Monday.